chore(STONEO11Y-75): Add limitations to README (#28)
redhat-appstudio/o11y: README.md

Reviewer: Can we break the lines at 89 characters or something similar?
Will make it easier to maintain.

Author: Done

# RHTAP Observability 

> This repository contains Prometheus recording rule files and alert rule files for observability use, as well as unit tests for these rules and alerts.

Reviewer: alerts are rules as well
```suggestion
This repository contains Prometheus recording rule files and alert rule files for observability use, as well as unit tests for these rules.
```

Author: Done

## Recording Rules

> - `appstudio_container_resource_limits` collects all the resources limits (cpu and memory) for all containers and init_containers, and adding the `label_pipelines_appstudio_openshift_io_type` label to our metric

Reviewer: ```suggestion
- `appstudio_container_resource_limits` collects all the resource limits (cpu and memory) for all containers and init_containers, and adds the `label_pipelines_appstudio_openshift_io_type` label to the new metric.
```

> - `appstudio_container_resource_minutes_gauge` uses the `appstudio_container_resource_limits` to create a 1 minutes gauge and calculate the limits according to the live time of the container

Reviewer 2: 1 minute (no 's').
Should "live time" be "lifetime" or "live-time" is a common expression?

Reviewer: `resources limits` -> `resource limits`

Author: Done

> - `appstudio_container_resource_minutes_gauge` uses the `appstudio_container_resource_limits` to create a 1 minutes gauge and calculate the limits according to the live time of the container

Reviewer: `and calculate` -> `and calculates`

> - `appstudio_container_network_transmit_bytes_total` collects the network egress of containers and adding the `label_pipelines_appstudio_openshift_io_type` label to our metric.

Author: Done

### Limitations

`appstudio_container_network_transmit_bytes_total` is using cAdvisor's `container_network_transmit_bytes_total` to collect the data.  

> We encounetered several issues while testing it with pods containing init_containers that transmit data.   

Reviewer 2: Did we resolve the issues we encountered? If so, should it be mentioned?

Reviewer: I think we should just mention the limitations rather than explaining the process we had.

Author: Done, I added the description of the limitation

# RHTAP Observability 

> This repository contains Prometheus recording rule files and alert rule files for  
> observability use, as well as unit tests for these rules.

Reviewer: ```suggestion
This repository contains Prometheus recording rule files and alert rule files for
observability use, as well as unit tests for these rules.
```

Author: Done, PTAL
